components/style/properties/longhand/font.mako.rs

```diff
@@ -1006,3 +1006,865 @@ ${helpers.single_keyword("font-variant-position",
+        let mut ornaments = None;
+        let mut annotation = None;
+
+        loop {
+            if let Ok(_) = input.try(|input| input.expect_ident_matching("historical-forms")) {
+                if historical_forms {
+                    return Err(());
+                }
+                historical_forms = true;
+            }
+
+            if let Ok(func_name) = input.expect_function() {
+                match_ignore_ascii_case! {&func_name,
+                    "styleset" => {
+                        if styleset.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let args = try!(input.parse_comma_separated(|input| {
+                                Ok(try!(input.expect_ident()).into_owned())
+                            }));
+                            styleset = Some(args);
+                            Ok(())
+                        }));
+                    },
+                    "character-variant" => {
+                        if character_variant.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let args = try!(input.parse_comma_separated(|input| {
+                                Ok(try!(input.expect_ident()).into_owned())
+                            }));
+                            character_variant = Some(args);
+                            Ok(())
+                        }));
+                    },
+                    "swash" => {
+                        if swash.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let arg = try!(input.expect_ident()).into_owned();
+                            swash = Some(arg);
+                            Ok(())
+                        }));
+                    },
+                    "ornaments" => {
+                        if ornaments.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let arg = try!(input.expect_ident()).into_owned();
+                            ornaments = Some(arg);
+                            Ok(())
+                        }));
+                    },
+                    "annotation" => {
+                        if annotation.is_some() {
+                            return Err(());
+                        }
+
+                        try!(input.parse_nested_block(|input| {
+                            let arg = try!(input.expect_ident()).into_owned();
+                            annotation = Some(arg);
+                            Ok(())
+                        }));
+                    },
+                    _ => break, 
```

Comment:
(I could be wrong, but) I think this is wrong. Right now, we break the loop if we find anything other than the expected names (for which we should actually return an error). Since `match_ignore_ascii_case!` recently added support for pattern guards, we should do something [like what we do](https://dxr.mozilla.org/servo/rev/f1ae547b9faf82c74f1680b1c5ff456b57d153c2/components/style/properties/longhand/position.mako.rs#381) for `grid-auto-flow`